src/main/frontend/handler/page.cljs

```diff
@@ -443,24 +406,55 @@
 
       (ui-handler/re-render-root!))))
 
+(defn- rename-nested-pages
+  [old-ns-name new-ns-name]
+  (when-let [nested-pages (db/get-nested-pages
+                           (state/get-current-repo)
+                           (string/lower-case old-ns-name))]
+    (doseq [page nested-pages]
+      (let [[_page-id old-page-name] page
+            new-page-name (string/replace old-page-name
+                                          (re-pattern (util/format "(?i)\\[\\[%s\\]\\]" old-ns-name))
+                                          (util/format "[[%s]]" new-ns-name))]
+        (rename-page-aux old-page-name new-page-name)))))
+
+(defn- rename-namespace-pages!
+  [repo old-name new-name]
+  (let [pages (db/get-namespace-pages repo old-name)]
+    (doseq [{:block/keys [name original-name] :as page} pages]
+      (let [old-page-title (or original-name name)
+            new-page-title (build-new-namespace-page-title old-page-title old-name new-name)]
+        (when (and old-page-title new-page-title)
+          (p/let [_ (rename-page-aux old-page-title new-page-title)]
+            (println "Renamed " old-page-title " to " new-page-title)))))))
+
 (defn rename!
   [old-name new-name]
-  (let [old-name      (string/trim old-name)
+  (let [repo          (state/get-current-repo)
+        old-name      (string/trim old-name)
         new-name      (string/trim new-name)
+        namespace     (or (string/includes? old-name "/")
+                          (db/get-namespace-pages repo old-name))
         name-changed? (not= old-name new-name)]
-    (when (and old-name
-               new-name
-               (not (string/blank? new-name))
-               name-changed?)
+    (if (and old-name
+             new-name
+             (not (string/blank? new-name))
+             name-changed?)
       (cond
-        (= (string/lower-case old-name) (string/lower-case new-name))
-        (rename-page-aux old-name new-name)
-
         (db/pull [:block/name (string/lower-case new-name)])
         (notification/show! "Page already exists!" :error)
 
         :else
-        (rename-page-aux old-name new-name)))))
+        (do (if namespace
+              (rename-namespace-pages! repo old-name new-name)
+              (rename-page-aux old-name new-name))
+            (rename-nested-pages old-name new-name)))
+      (cond
+        (string/blank? new-name)
+        (notification/show! "Please use a valid name, empty name is not allowed!" :error)
+
+        (not name-changed?)
+        (notification/show! "Cannot rename to page with same same!" :error)))))
```

Comment:
typo, same name!

Reply:
Add an `:else` catch all branch always when using `cond` to make the code more robust.

Reply:
> Add an `:else` catch all branch always when using `cond` to make the code more robust.

You are right!